apps/src/lab2/projects/ProjectManager.ts

```diff
@@ -91,7 +91,11 @@ export default class ProjectManager {
     }
 
     this.lastChannel = channel;
-    return {sources, channel};
+    const response = await HttpClient.fetchJson<{abuse_score: number}>(
+      `/v3/channels/${this.channelId}/abuse`
+    );
+    const abuseScore = response.value.abuse_score;
```

Comment:
could we hit a null reference here?

Reply:
This route points to `show_abuse` in `report_abuse_controller` which returns the abuse score from [`projects.rb` `get_abuse`](https://github.com/code-dot-org/code-dot-org/blob/050f0bf14eab9fdf1df087d18c2c16aa9ecd0cba/dashboard/legacy/middleware/helpers/projects.rb#L420-L424). When a [project is created](https://github.com/code-dot-org/code-dot-org/blob/050f0bf14eab9fdf1df087d18c2c16aa9ecd0cba/dashboard/legacy/middleware/helpers/projects.rb#L32-L49), the `abuse_score` is set to `0`. If the channel id is not found, then an error will be thrown [in `show_abuse`](https://github.com/code-dot-org/code-dot-org/blob/050f0bf14eab9fdf1df087d18c2c16aa9ecd0cba/dashboard/app/controllers/report_abuse_controller.rb#L72-L74).
So I don't believe we'll hit a null reference here. Lmk if you agree!

Reply:
makes sense!